src/adaptation/Adaptation.jl

```diff
@@ -1,75 +1,59 @@
 module Adaptation
+export Adaptation
 
-import Base: string, rand
-using Random: GLOBAL_RNG, AbstractRNG
-using LinearAlgebra: Symmetric, UpperTriangular, mul!, ldiv!, dot, I, diag, cholesky, UniformScaling
-import LinearAlgebra, Statistics
-using ..AdvancedHMC: DEBUG
+using LinearAlgebra: LinearAlgebra
+using Statistics: Statistics
 using Parameters: @unpack, @pack!
 
-const AbstractScalarOrVec{T} = Union{T,AbstractVector{T}} where {T<:AbstractFloat}
+using ..AdvancedHMC: DEBUG, AbstractScalarOrVec
 
 abstract type AbstractAdaptor end
-
-##
-## Interface for adaptors
-##
-
-getM⁻¹(adaptor::T) where {T<:AbstractAdaptor} = error("`getM⁻¹(adaptor::$T)` is not implemented.")
-getϵ(adaptor::T) where {T<:AbstractAdaptor} = error("`getϵ(adaptor::$T)` is not implemented.")
-adapt!(
-    adaptor::T,
-    θ::AbstractVecOrMat{<:AbstractFloat},
-    α::AbstractScalarOrVec{<:AbstractFloat}
-) where {T<:AbstractAdaptor} = error("`adapt!(adaptor::$T, θ::AbstractVecOrMat{<:AbstractFloat}, α::AbstractScalarOrVec{<:AbstractFloat})` is not implemented.")
-reset!(adaptor::T) where {T<:AbstractAdaptor} = error("`reset!(adaptor::$T)` is not implemented.")
-initialize!(adaptor::T, n_adapts::Int) where {T<:AbstractAdaptor} = throw(MethodError(initialize!, adaptor, n_adapts))
-finalize!(adaptor::T) where {T<:AbstractAdaptor} = error("`finalize!(adaptor::$T)` is not implemented.")
+function getM⁻¹ end
+function getϵ end
+function adapt! end
+function reset! end
+function initialize! end
+function finalize! end
+export AbstractAdaptor, adapt!, initialize!, finalize!, reset!, getϵ, getM⁻¹
 
 struct NoAdaptation <: AbstractAdaptor end
-
+export NoAdaptation
 include("stepsize.jl")
+export StepSizeAdaptor, NesterovDualAveraging
 include("precond.jl")
+export MassMatrixAdaptor, UnitMassMatrix, WelfordVar, WelfordCov
+include("composite.jl")
+export CompositeAdaptor, NaiveHMCAdaptor, StanHMCAdaptor
+
+# User helpers for AdvancedHMC
+
+using ..AdvancedHMC: 
+    AbstractIntegrator, nom_step_size,
+    AbstractMetric, UnitEuclideanMetric, DiagEuclideanMetric, DenseEuclideanMetric
 
-# TODO: implement consensus adaptor
+StepSizeAdaptor(δ::AbstractFloat, i::AbstractIntegrator) = 
+    NesterovDualAveraging(δ, nom_step_size(i))
 
-##
-## Compositional adaptor
-## TODO: generalise this to a list of adaptors
-##
+@deprecate NesterovDualAveraging(δ, i::AbstractIntegrator) StepSizeAdaptor(δ, i)
 
-struct NaiveHMCAdaptor{M<:AbstractPreconditioner, Tssa<:StepSizeAdaptor} <: AbstractAdaptor
-    pc  :: M
-    ssa :: Tssa
-end
-Base.show(io::IO, a::NaiveHMCAdaptor) = print(io, "NaiveHMCAdaptor(pc=$(a.pc), ssa=$(a.ssa))")
+MassMatrixAdaptor(m::UnitEuclideanMetric{T}) where {T} = 
+    UnitMassMatrix(T)
```

Comment:
IMO
```suggestion
    UnitMassMatrix{T}()
```
feels more natural if `T` is just a type parameter.

Reply:
Good point.